File: modulefederation/apps/web/src/app/portal/model/microfrontend-version.ts
import { Manifest } from "@modulefederation/portal";
import { MicrofrontendInstance } from "./microfrontend-instance";

export interface MicrofrontendVersion {
    id: string,
    microfrontend: string,
    version: string,
    type: string,
    manifest : Manifest,
    enabled : boolean,
    configuration : string,
    instances : MicrofrontendInstance[],
    applicationVersion: number | null
}

File: portal/core/src/main/java/com/serious/portal/model/MicrofrontendVersion.kt
package com.serious.portal.model
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

data class MicrofrontendVersion(
    var id: String, // combination of mfe:version
    var microfrontend: String,
    var version: String,
    var manifest : Manifest,
    var configuration : String,
    var enabled : Boolean,
    var instances : List<MicrofrontendInstance>
)